### ISSUE TITLE
HTML Help Workshop rate limited archive.org

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags-manual#148]
The link used by the chocolatey package has been removed from archive.org (!)

Need to fix the link from chocolatey to somewhere else or just store the binaries somewhere we can retrieve. If someone has the previous download, please push somewhere and update the link below.

https://github.com/dwickern/chocolatey-packages/tree/master/html-help-workshop

Alternatively, switch totally for the freepascal port.

Even better alternative is to build a tiny webviewer (using the webviewer already there in . NET) with the index/search functionality and replace the help system in the Editor - in my Windows 10 I can no longer view chm files I download alone. The unblock button is gone from them.

**Comment by morganwillcock** [comment]
> The link used by the chocolatey package has been removed from archive.org (!)

I think it was probably just rate-limiting, the download is still there.

> Alternatively, switch totally for the freepascal port.

I have this working but:
- the internal search index (generated by the compiler) is smaller than the current one, I think it doesn't allow searching based on partial word matches
- there is a similar problem to get the chmcmd binaries onto the system because pulling them from source is likely deemed abusive towards Sourceforge

Possibly we can have a volatile repository that builds to GitHub Pages and has the chmcmd  binaries and the current compiler, and the current compiler can be encrypted with a CI token so that we are not 'distributing' it.

> Even better alternative is to build a tiny webviewer (using the webviewer already there in . NET) with the index/search functionality and replace the help system in the Editor

As long as it is accepted in terms of panel/window behaviour it would probably be OK. But given the Editor itself doesn't have good panel and window behaviour right now I'm not sure it is a good idea to build on top of that. Display scaling might be an issue too.

> in my Windows 10 I can no longer view chm files I download alone. The unblock button is gone from them.

Possibly it depends on where the files are and/or whether they are writeable. But you can unblock them in recent Powershell versions with [Unblock-File](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/unblock-file).

**Comment by ericoporto** [comment]
> Possibly we can have a volatile repository that builds to GitHub Pages and has the chmcmd binaries and the current compiler, and the current compiler can be encrypted with a CI token so that we are not 'distributing' it.

We could create a private repository in the AGS org for this and use a token to get it from it's [release](https://stackoverflow.com/questions/20396329/how-to-download-github-release-from-private-repo-using-command-line). A bit ugly but at least it could work.

**Comment by ericoporto** [comment]
I learned that GitHub Actions has a Cache https://github.com/actions/cache

Apparently we could cache the dependencies after a successful fetch of these and just pick things from the cache directly.

**Comment by morganwillcock** [comment]
There is a separate ticket open for the caching so I'll close this issue which was more about the download failing for this one particular file. If continuing along the CHM path then switching to chmcmd is possible, the new build system has an option to switch to it. The only drawback is that the generated search index from chmcmd seems smaller - I think searching for partial words would give fewer results.
